### DISCUSSION TITLE
False positives for security fixes part of .NET 8.0.10

### DISCUSSION THREAD

**Discussion by raoganeshr** [aquasecurity/trivy#7693] [False Detection]
### IDs

CVE-2024-43485, CVE-2024-43483

### Description

https://msrc.microsoft.com/update-guide/vulnerability/CVE-2024-43485 & https://avd.aquasec.com/nvd/2024/cve-2024-43483 are fixed in .net 8.0.10. However the nuget packages for the associated fixes are not numbered as 8.0.10. Hence even after upgrading to the patched versions, we are unable to fix the trivy findings. Trivy database needs to be updated to correct the fixed package versions for these two nuget packages.

1. System.Text.Json (CVE-2024-43485)
https://github.com/advisories/GHSA-8g4q-xg66-9fp4 - Has correct version
![image](https://github.com/user-attachments/assets/f8c2b961-e42b-430c-8f2b-fd6562c33d8f)
https://www.nuget.org/packages/system.text.json/
![image](https://github.com/user-attachments/assets/ecc012dd-6cc8-4445-b3ab-bd7153080d50)

2. System.Runtime.Caching (CVE-2024-43483)
3. https://github.com/advisories/GHSA-qj66-m88j-hmgj - Has correct version
![image](https://github.com/user-attachments/assets/b777ddd4-5f71-4dcc-a7d8-b069403934c8)
https://www.nuget.org/packages/System.Runtime.Caching/9.0.0-rc.2.24473.5
![image](https://github.com/user-attachments/assets/db567cd9-d071-48f2-a964-ed223864e484)

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Filesystem

### Scanner

Vulnerability

### Debug Output

```bash
N.A.
```

### Version

```bash
Version: 0.56.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-10-09 00:24:02.775946363 +0000 UTC
  NextUpdate: 2024-10-10 00:24:02.775946233 +0000 UTC
  DownloadedAt: 2024-10-09 00:31:07.910605636 +0000 UTC
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment]
Hello @raoganeshr 
Thanks for your report!

>System.Text.Json (https://github.com/advisories/GHSA-8g4q-xg66-9fp4)
https://github.com/advisories/GHSA-8g4q-xg66-9fp4 - Has correct version

Can you update trivy-db?
Fixed versions are correct now:
![изображение](https://github.com/user-attachments/assets/e5cba45c-24fc-47a8-a510-cf420e548266)

> System.Runtime.Caching (https://github.com/advisories/GHSA-qj66-m88j-hmgj)
https://github.com/advisories/GHSA-qj66-m88j-hmgj - Has correct version

same.

> However the nuget packages for the associated fixes are not numbered as 8.0.10

Do you mean `System.Text.Json` (as an example) has version `8.0.4` before and after `.net` upgrade to `8.0.10`?

**Reply by raoganeshr** [reply]
No. I meant even though the CVE advisory mentioned that its fixed in .NET 8.0.10, the nuget packages dont follow that versioning. I believe the updated GitHub Advisory has the correct versions for the fixed components. I will proceed with updating the trivy database. Thanks for your help.
